WordPress/Jetpack/Classes/System/JetpackWindowManager.swift

```diff
@@ -73,12 +80,43 @@ class JetpackWindowManager: WindowManager {
         self.show(container.makeInitialViewController())
     }
 
-    private func switchToAppUI(for blog: Blog?) {
+    func switchToAppUI(for blog: Blog?) {
         cancellable = nil
         showAppUI(for: blog)
     }
 
-    private var shouldShowMigrationUI: Bool {
-        return FeatureFlag.contentMigration.enabled && AccountHelper.isLoggedIn
+    func handleMigrationFailure(_ error: DataMigrationError) {
+        guard
+            case .dataNotReadyToImport = error,
+            isCompatibleWordPressAppPresent,
+            let schemeUrl = URL(string: AppScheme.wordpressMigrationV1.rawValue)
+        else {
+            showSignInUI()
+            return
+        }
+
+        let wpInstallationState = MigrationAppDetection.getWordPressInstallationState()
+        MigratableStateTracker().track(wpInstallationState)
+
+        /// The WordPress pre-flight process hasn't ran, but WordPress is installed.
+        /// Note: We don't know if the user has ever logged into WordPress at this point, only
+        /// that they have a version compatible with migrating.
+        let actions = MigrationLoadWordPressViewModel.Actions()
+        let loadWordPressViewModel = MigrationLoadWordPressViewModel(actions: actions)
+        let loadWordPressViewController = MigrationLoadWordPressViewController(viewModel: loadWordPressViewModel)
+        actions.primary = {
+            UIApplication.shared.open(schemeUrl) { [weak self] _ in
+                loadWordPressViewController.dismiss(animated: true) {
+                    self?.showSignInUI()
+                }
+            }
+        }
+        actions.secondary = { [weak self] in
+            loadWordPressViewController.dismiss(animated: true) {
+                self?.showSignInUI()
+            }
+        }
+        self.show(loadWordPressViewController)
+        return
```

Comment:
I suppose we don't need this `return`?

Reply:
Nope! It was wiped with the method extraction. 🙇